### PR TITLE
Polish the generic STM32 instructions

### PR DESCRIPTION
**Pull request by tronical**
This includes removal of the SLINT_STYLE setting, as a default is set there for FEATURE_FREESTANDING.

### CODE REVIEW

File: api/cpp/docs/mcu/stm32/generic.md [hunk 1]
```diff
@@ -12,30 +12,95 @@ as well as the peripherals of our board.
...
+                text: "Hello World";
+                font-size: 18px;
+                horizontal-alignment: center;
+            }
+        }
+    }
+    ```
+14. Add a `.slint` file to your project and include it in your project by calling `slint_target_sources`:
+    ```cmake
+    slint_target_sources(your-target app-window.slint)
```
Comment (ogoffart) [inline api/cpp/docs/mcu/stm32/generic.md:62]: this step could be merged in step 11 so there is only one copy-paste for this

File: api/cpp/docs/mcu/stm32/generic.md [hunk 2]
```diff
@@ -12,30 +12,95 @@ as well as the peripherals of our board.
...
+
+        TS_Init_t hTS;
+        hTS.Width = LCD_DEFAULT_WIDTH;
+        hTS.Height = LCD_DEFAULT_HEIGHT;
+        hTS.Orientation = TS_SWAP_XY;
+        hTS.Accuracy = 0;
+        /* Touchscreen initialization */
+        if (BSP_TS_Init(0, &hTS) != 0) {
+            Error_Handler();
+        }
```
Comment (ogoffart) [inline api/cpp/docs/mcu/stm32/generic.md:94]: I wonder how much of this should be done in slint_stm32_init instead.

File: api/cpp/docs/mcu/stm32/generic.md [hunk 3]
```diff
@@ -12,30 +12,95 @@ as well as the peripherals of our board.
...
-11. Add a C++ source file to your project, for example `appmain.cpp`, with an `appmain` function and call it
+8. Open a web browser, navigate to <https://github.com/slint-ui/slint/releases/latest>, and download
+   and extract `Slint-cpp-VERSION-thumbv7em-none-eabihf.tar.gz`, replace `VERSION` with the version you see.
+10. Set `CMAKE_PREFIX_PATH` to the extracted directory.
+11. Adjust your `CMakeLists.txt` for use of Slint. Copy the follow snippets and adjust the target names as needed:
+    ```cmake
+    # Locate Slint
+    find_package(Slint)
+    
+    # Default to pre-compiling images and fonts
```
Comment (ogoffart) [inline api/cpp/docs/mcu/stm32/generic.md:30]:
```suggestion
    # Embeds images and fonts in the binary
```

**Review by ogoffart — APPROVED** [review]

File: api/cpp/docs/mcu/stm32/generic.md (hunk 2 above)
Comment (tronical) [reply api/cpp/docs/mcu/stm32/generic.md:95]: Be my guest :-)

**Pull request merged**